prereise/gather/griddata/hifld/calculate/remap.py

```diff
@@ -0,0 +1,60 @@
+import copy
+
+
+def get_zone_mapping(zone):
+    """Generate dictionaries of zone using the zone.csv
+
+    :param pandas.DataFrame zone: the zone.csv data
+    :return: (*tuple*) -- a tuple of two dictionaries, where the first
+    dictionary is a dict mapping the STATE to its ID and the second is a dict
+    mapping a tuple of state and interconnect to its ID.
+    """
+
+    zone_dic = {}
+    zone_interconnect_dic = {}
+    for i in range(len(zone)):
+        tu = (zone["zone_name"][i], zone["interconnect"][i])
+        zone_dic[zone["zone_name"][i]] = zone["zone_id"][i]
+        zone_interconnect_dic[tu] = zone["zone_id"][i]
+    return zone_dic, zone_interconnect_dic
```

Comment:
I would do:
```
if zone.index.name != "zone_id":
    zone.set_index("zone_id", inplace=True)
return {v: k for k, v in toto["zone_name"].to_dict().items()}, zone.groupby(["zone_name", "interconnect"]).groups
```

Reply:
@BainanXia and @danielolsen, do we need to return the second element now that the `zone_name` are unique?

Reply:
We don't seem to use the second element at all, at least so far. These functions are generating lots of mappings, but it's unclear what the mappings will be used for. Once we refactor to use built-in pandas functions, if these functions are only used in one place, I'm not sure it's even worth defining them rather than just including the logic in the higher-level function.

Reply:
@rouille I agree with @danielolsen, these returns contain duplicated information given we have one-to-one mapping between `zone_id` and `zone_name`. Also, we should be clear what mapping is useful during grid model creation to avoid duplicate code with `powersimdata.network.hifld.constants.zones.py` in the future.